### ISSUE TITLE
[🐛 Bug]: CdpVersionFinder does not find any implementation for Spring Boot Application

### ISSUE THREAD

**Issue by globus1987** [SeleniumHQ/selenium#10496]
### What happened?

We have a framework which uses Selenium 4 java library.
Our Application uses Spring Boot.
Execution from both Intellij and Gradle works well but this is because classpath is accessed directly.
When we create a fat jar, ServiceLoader cannot access inner libraries to load them and use.
It results with NoOp and disability to create session

### How can we reproduce the issue?

```shell
Create Spring boot application which wants to open browser using selenium.
Add line to create CDP session afterwards.
Build FatJar, run from jar file using java -jar
```

### Relevant log output

```shell
---
```

### Operating System

Windows 10, Linux

### Selenium version

4.1.3

### What are the browser(s) and version(s) where you see this issue?

Chrome 99

### What are the browser driver(s) and version(s) where you see this issue?

ChromeDriver according to 99 version (WebDriverManager)

### Are you using Selenium Grid?

No

**Comment by diemol** [comment]
Please share a GitHub repository we can use to reproduce the issue.

**Comment by dennisoelkers** [comment]
@globus1987: Maybe this is fixed with #10509?

**Comment by diemol** [comment]
Good point @dennisoelkers, it does make sense.

Will close this and then please re-test when 4.1.4 or 4.2.0 are out.
